module/modmaninterface/interface.go

```diff
@@ -11,7 +11,7 @@ import (
 
 // ModuleManager abstracts the module manager interface.
 type ModuleManager interface {
-	Add(ctx context.Context, cfg config.Module) error
+	Add(ctx context.Context, confs ...config.Module) error
```

Comment:
beyond the scope of this PR, but I think we should move to a collection based interface for the other methods as well. this gives us more flexibility over implementation details such whether to operate in parallel or not

Reply:
It's an interesting idea; which other methods in the interface are you referring to?